pandas/core/indexes/multi.py

```diff
@@ -3095,6 +3105,32 @@ def _update_indexer(idxr, indexer=indexer):
         # empty indexer
         if indexer is None:
             return Int64Index([])._ndarray_values
+
+        # Generate tuples of keys by which to order the results
```

Comment:
this is really complex and adding quite a bit of code. Please take another look to simplify greatly.

Reply:
Sure thing, will do. I did have a simpler solution, but the performance hit was really high.